server/middleware/token.js

```diff
@@ -32,6 +32,9 @@ function rewriteUserLiteral(req, currentUserLiteral) {
       debug('req.url has been rewritten from %s to %s', urlBeforeRewrite,
         req.url);
     }
+  } else if (!req.accessToken && currentUserLiteral &&
+      new RegExp(currentUserLiteral).test(req.url)) {
```

Comment:
This will match requests that do not contain current user literal. Assuming the literal is `me`, this will match requests like `/products/watermelon` too.

My proposal: use the same regexp as on L29.

```js
function rewriteUserLiteral(req, currentUserLiteral) {
  if (!currentUserLiteral) return;
  const literalRegExp = new RegExp('/' + currentUserLiteral + '(/|$|\\?)', 'g');
  // ...
}